howto/port/install-as-system-app.md

```diff
@@ -0,0 +1,86 @@
+3. Include the addon in an application after adding it to AMS.
+4. Enable the `allow_custom_system_signatures` feature in the application manifest file.
+5. Create the application.
+
+When the application is created successfully, the APK will be installed as a system app in the Android container.
+
+### Make a system app
+
+To build your app as a system app instead of a user app, add the attribute `android:sharedUserId="android.uid.system"` into the `<manifest>` tag in the `AndroidManifest.xml` file of your Android app. This attribute will allow the app to run a process with system privileges.
+
+```xml
+...
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    package="<package_name>"
+    android:sharedUserId="android.uid.system">
+...
+```
+
+Then build and sign the application alongside with other Android applications when building your Android image. Alternatively, sign it with [Android Studio](https://developer.android.com/studio/publish/app-signing). This method does not require the system platform key. Instead, you can use the keys that are generated from Android Studio to sign the application.
+
+### Create an addon
+
+To install the signed APK as a system app in the Android container, create an addon and invoke the `aam install-system-app` command in a `pre-start` hook.
+
+Follow the [tutorial](https://discourse.ubuntu.com/t/creating-an-addon/25284) to create the basic layout of an addon, with a `manifest.yaml` file and a `hooks` folder. Place the APK into the addon folder and create a `pre-start` hook with the following content (assuming that the APK file is named `app.apk`):
+
+```bash
+#!/bin/bash -ex
+
+# Only install the APK as a system app when bootstrapping an application.
+if  [ "$CONTAINER_TYPE" = "regular" ]; then
+  exit 0
+fi
+
+aam install-system-app \
+  --apk="${ADDON_DIR}"/app.apk \
+  --permissions=<comma-separated list of permissions that the application requires> \
+  --package-name=<package_name>
+```
+
+The values of the `package_name` and the `permissions` parameters must match the ones defined in the `AndroidManifest.xml` file of the Android project. If the app requires access to hidden Android APIs to function, add the `--access-hidden-api` parameter to the above command. Use `aam install-system-app --help` for details about this command.
+
+The final layout of the addon should look as follows:
+
+```bash
+.
+├── app.apk
+├── hooks
+│   └── pre-start
+└── manifest.yaml
+```
+
+Add this addon to AMS with the following command:
+
+    amc addon add install-sys-app .
+
+### Include the addon in an application
+
+To use this addon in an application, include the addon name under `addons` in the application manifest file when creating an application. You must also enable the feature `allow_custom_system_signatures`, which ensures that the `aam install-sys-app` command that is invoked in the `pre-start` hook of the addon works properly.
```

Comment:
which ensures that the `aam install-sys-app` command
->
which ensures that the `aam install-system-app` command